server/src/main/java/org/candlepin/model/CdnCurator.java

```diff
@@ -35,6 +35,8 @@ public CdnCurator() {
      * @return CDN whose label matches the given value.
      */
     public Cdn getByLabel(String label) {
+        // TODO: This is dangerous. We're expecting a unique result, but there is no guarantee the
```

Comment:
I think `uniqueResult` will throw an exception if it's not unique (both in [Hibernate](https://docs.jboss.org/hibernate/orm/5.1/javadocs/org/hibernate/Query.html#uniqueResult--) and in [JPA's version](https://docs.oracle.com/javaee/7/api/javax/persistence/Query.html#getSingleResult--)).  So we can catch the exception and do something or just let a runtime error get thrown.

Reply:
We can, but the problem isn't the lack of exception handling, it's that we expect a non-unique field to be unique. If someone's deployment ends up with two Cdns with the same label, they're going to have issues regardless of whether or not we catch that exception. This is something that we should probably fix by adding a DB restriction, or allowing more than a single object to be output from this method.